### ISSUE TITLE
Onfido framework size is inexplicably huge: 94.2MB

### ISSUE THREAD

**Issue by acecilia** [onfido/onfido-ios-sdk#364]
### What was the expected behaviour?

Onfido iOS SDK to have a reasonable size

### What happened instead?

Latest version `28.3.0` has a framework size for `arm64` of `94.2MB`, and a binary size of `88MB`. 

Inside an app this translates to a framework size of `30MB` and a binary size of `24MB`.

### Version info:

- Onfido SDK (Debug/Release): `28.3.0`

### Steps to reproduce:

Download prebuilt xcframework from [here](https://github.com/onfido/onfido-ios-sdk/releases/download/28.3.1/Onfido-v28.3.0.zip) and inspect it

**Comment by acecilia** [comment]
Onfido framework includes multiple json files that in total amount to 1.6MB in the final app. The bulk of this size is the file `supported-documents.json`.

These json files are not [minified](https://codebeautify.org/jsonminifier). 

As an example: minifying the file `supported-documents.json` reduces its size from 1.6MB to 1MB, a 37% reduction

**Comment by acecilia** [comment]
Onfido includes 2.4MB of localized translations. These translations are not [minified](https://docs.emergetools.com/docs/minify-localized-strings). 

Minifying them can have an estimated 90% reduction in size

**Comment by acecilia** [comment]
In `28.1.0`, Onfido includes a binary that amounts to 24,2MB. The binary contains unstripped symbols.

Following [this](https://docs.emergetools.com/docs/strip-binary-symbols) steps and executing the following on the binary:
```
strip -rSTx Onfido.framework/Onfido
```
Reduces the binary size from 24.2MB to 13.6MB, A reduction of 43%

**Comment by vguerci** [comment]
Thank you @acecilia for the detailed issue, investigation and knowledge sharing, very helpful 🙏

We will consider stripping binaries the way you are proposing in upcoming versions.

Is this still an issue for you or are you OK with the 13MB reduced binary?

**Comment by acecilia** [comment]
Anything you can do to reduce the SDK size is more than welcome, thank you!

> Is this still an issue for you or are you OK with the 13MB reduced binary?

I can provide more feedback as soon as I get a new version of the SDK🙏 

The best solution from my point of view is to distribute both dynamic and static versions of the framework, so consumers can decide which and how to integrate them. Static linking benefits not only size, but also app startup time

**Comment by vguerci** [comment]
From what I see, the size reduction mostly comes from removing debug symbols, which I guess will make desymbolication impossible. Do you confirm this, also, is this something you are OK with?

We are also considering packaging as a static framework, but that is lower in our priorities as it is a relatively complex task.

**Comment by vguerci** [comment]
With dSYMs, we have the ability to desymbolicate customers' crash reports with our source code. And at first sight, dSYMs are affected by `strip` `-S` "Remove the debugging symbol table entries" option. It would require more tests to confirm though.

I wished our build pipelines were that simple, but sadly packaging our SDK is a bit more complex than this.

**Comment by acecilia** [comment]
> And at first sight, dSYMs are affected by strip -S "Remove the debugging symbol table entries" option

I do not think so. From [here](https://www.emergetools.com/blog/posts/how-xcode14-unintentionally-increases-app-size):
> Binary symbol stripping is where certain types of metadata that are unnecessary for running the app in production are removed from the binary. This is metadata that can be helpful prior to production, e.g. to generate dSYM files with, but only adds bloat to a user's phone in a production build.

> I wished our build pipelines were that simple, but sadly packaging our SDK is a bit more complex than this.

Feel free to reach out privately (I have already a case opened with Onfido support, you can use that or I can give you my email). I may be able to advise a solution if you were able to provide a high level description of your pipeline

**Comment by abozaidibrahimON** [comment]
@acecilia we are going to remove the `supported-documents.json` in the next version, it is in development now.
